src/combinatorics/simple_graph/basic.lean

```diff
@@ -640,6 +640,30 @@ lemma delete_edges_eq_inter_edge_set (s : set (sym2 V)) :
   G.delete_edges s = G.delete_edges (s ∩ G.edge_set) :=
 by { ext, simp [imp_false] { contextual := tt } }
 
+/-! ## Induced graphs -/
+
+/- Given a set `s` of vertices, we can restrict a graph to those vertices by restricting its
+adjacency relation. This gives a map between `simple_graph V` and `simple_graph s`.
+
+There is also a notion of induced subgraphs (see `simple_graph.subgraph.induce`). -/
+
+/-- Restrict a graph to the vertices in the set `s`. -/
+@[simps]
+def induce (s : set V) (G : simple_graph V) : simple_graph s :=
+{ adj := λ u v, G.adj u v }
+
+/-- Given a graph on a set of vertices, we can make it be a `simple_graph V` by
+adding in the remaining vertices without adding in any additional edges. -/
+@[simps]
+def spanning_coe {s : set V} (G : simple_graph s) : simple_graph V :=
+{ adj := λ u v, ∃ (hu : u ∈ s) (hv : v ∈ s), G.adj ⟨u, hu⟩ ⟨v, hv⟩ }
```

Comment:
Isn't this precisely the image under `subtype.val : ↥s → V`, but very slightly better definitional properties? I'm not sure it deserves a separate definition.

Reply:
Could you be more specific here? If we have images of graphs, I've missed it. One complexity with graph images is that we have to exclude loop edges specially when the function isn't an injection.

This could be defined using `relation.map`, but it's true that the way it's written it has better definitional properties.

Reply:
This PR now includes `simple_graph.map` and `simple_graph.comap`, so it's become possible to follow your suggestion.